.github/workflows/generate-docs.yml

```diff
@@ -0,0 +1,60 @@
+name: generate-docs
+on: 
+  workflow_dispatch:
+jobs:
+  publish_docs:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install Build Dependencies
+        description: This step installs build dependencies for python-sdk and pdoc
+        run: |
+          apt install -y git \
+            python2 \
+            python2-dev \
+            curl \
+            asciidoctor \
+            gcc \
+            libxml2-dev \
+            libcurl4-openssl-dev \
+            libcurl4-gnutls-dev \
+            libexpat1-dev \
+            gettext \
+            libz-dev \
+            libssl-dev \
+            dev-libgnutls28
+
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install Python-SDK with Python 2.7
+      - description: This step installs the python-sdk with python2.7 in order to enable a \
+                       version of pdoc, which doesn't support python3, to work with it. 
+        run: |
+          .automation/.generate-setup-files.sh
+          python2.7 setup.py install --user --prefix=
+        shell: bash
+
+      - name: Install pip2.7
+      - description: This step installs pip2.7, which will be needed for the installation of pdoc.
+        run: |
+          curl https://bootstrap.pypa.io/pip/2.7/get-pip.py --output get-pip.py 
+          python2 get-pip.py 
+
+      - name: Install Pdoc
+      - description: This steps installs an extension of pdoc which supports asciidoc and runs on python2.
+      - run: |
+          git clone https://github.com/machacekondra/pdoc
+          cd pdoc
+          pip2.7 install -U .
+
+      - name: Install pygments
+      - run: pip2.7 install pygments
+
+      - name: Generate documentation
+      - run: pdoc ovirtsdk4 --html --html-dir=html
```

Comment:
Documentation needs to be generated as a part of build workflow, please integrate this workflow into the build one so documentation can be properly generated as a part of the build:

https://github.com/oVirt/python-ovirt-engine-sdk4/blob/main/.github/workflows/build.yaml#L170

Reply:
Off course, getting there. I should have marked it as WIP